presto-main/src/main/java/com/facebook/presto/server/remotetask/HttpRemoteTask.java

```diff
@@ -637,6 +660,22 @@ private int getPendingSourceSplitCount()
         return pendingSourceSplitCount;
     }
 
+    public synchronized boolean isAllSplitsRun()
+    {
+        return unprocessedSplits.values().stream().allMatch(Map::isEmpty);
+    }
+
+    public synchronized boolean isOnlyOneSplitLeft(PlanNodeId planNodeId)
```

Comment:
IS this function name accurate ? It seems to check for there being only one scan node instead of one split in that scan node.

Reply:
Can you also help me understand this limitation better. Why do we have it ?